lib/tracking_options.dart

```diff
@@ -0,0 +1,114 @@
+import 'constants.dart';
+
+/// Disable tracking of user properties.
+///
+/// Before initialzing Amplitude instance, create a TrackingOptions instance
+/// with your desired tracking options and pass it to the Amplitude instance.
+class TrackingOptions {
+  Set<String> _disabledFields = {};
+
+  // Available on iOS, Android, and Web
+  TrackingOptions disableIpAddress() {
+    _disabledFields.add(Constants.ampTrackingOptionIpAddress);
+    return this;
+  }
```

Comment:
Hi @justin-fiedler, I really like your suggestion to extract platform specific options for clarity and readability. I am considering doing the same for tracking options. 

Currently customers have to create an instance of `TrackingOptions` and pass the instance to configuration. It is how the next-gen SDKs work, for example
```
var trackingOptions = TrackingOptions().disableADID().disableApiLevel();
var configuration = Configuration(
  apiKey: "apiKey",
  trackingOptions: TrackingOptions({disableIpAddress: true})
);
```

Wdyt if we change `trackingOptions` to be `defaultTracking` style by changing `disablexxxx()` to a class property  so that we can use the same pattern to extract platform specific options out? For example
```
class TrackingOptions {
  final bool disableIpAddress;
  final AndroidTrackingOptions androidTrackingOptions;
}

class AndroidTrackingOptions {
 final bool disableADID;
}

var configuration = Configuration(
  apiKey: "apiKey",
  trackingOptions: TrackingOptions(
      disableIpAddress: true,
      androidTrackingOptions: AndroidTrackingOptions(disableADID: true)
  ) 
);
```

Reply:
After rethinking this, I feel it's better to keep it aligned with Kotlin and Swift SDK.

Reply:
It's not scalable as it's gonna be a breaking change if we add `disableLatLag` (currently only available on Android SDK) to iOS SDK.